From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Fri, 4 Nov 2016 18:57:49 -0400
Subject: curses: Report timeout.

* pinentry/pinentry-curses.c (dialog_run): Report if canceled due to
timeout.

Signed-off-by: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
---
 pinentry/pinentry-curses.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pinentry/pinentry-curses.c b/pinentry/pinentry-curses.c
index 9882cbf..a6dbb69 100644
--- a/pinentry/pinentry-curses.c
+++ b/pinentry/pinentry-curses.c
@@ -941,6 +941,7 @@ dialog_run (pinentry_t pinentry, const char *tty_name, const char *tty_type)
       if (timed_out && no_input)
 	{
 	  done = -2;
+          pinentry->specific_err = gpg_error (GPG_ERR_TIMEOUT);
 	  break;
 	}
 #endif
